tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/LiferayWorkspaceUtil.java

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core.workspace;
+
+import com.liferay.ide.core.LiferayWorkspaceNature;
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.core.util.FileUtil;
+
+import java.io.File;
+
+import org.eclipse.core.resources.IProject;
+
+/**
+ * @author Andy Wu
+ */
+public class LiferayWorkspaceUtil
+{
+
+    public static boolean isValidWorkspaceLocation( String location )
+    {
+        boolean retval = false;
+
+        File workspaceDir = new File( location );
+
+        File buildGradle = new File( workspaceDir, "build.gradle" );
```

Comment:
Lets also check to make sure that the workspace has the following folders;
modules
themes
but we also need to read the gradle.properties to look for custom names for those folders.  Look here:
https://github.com/david-truong/liferay-workspace/blob/master/gradle.properties#L3-L5

So we need to read those properties to look for custom names of "modules" and "themes"